### ISSUE TITLE
Wallet can miss processing spend txs in the pool

### ISSUE THREAD

**Issue by j-berman** [monero-project/monero#8670]
When the user has a spend tx in the pool that their wallet hasn't scanned yet (e.g. if the user rescans the chain, or uses the same seed in a different wallet), the wallet won't process the spend tx correctly. The user could end up trying to re-spend output(s) in this case and the daemon will reject the tx as a double spend.

The wallet process receive txs as expected. This issue appears isolated to spends.

**Comment by afungible** [comment]
I have noticed this behavior a couple months ago and I can confirm this occured. This results in double spend that the daemon rejects. I used same wallet file twice and used two monero-cli instances while fiddling around, on same PC.

Note, this also sort of corrupted my LMDB database (i.e. it showed a double spend txn) but I couldn't repair the DB locally. The only way was to download the whole Blockchain again, as whenever I spent from wallets outputs, it randomly popped up outputs are being double spent (when I clearly wasn't then) and was annoying.

Basically, one issue followed another in such a case.

**Comment by selsta** [comment]
Should be resolved by https://github.com/monero-project/monero/pull/10083.
